### PR TITLE
Getting started with Python input function and loops

### LINKED ISSUE #2340: User Input and while Loops in Python
### Proposed title of article
User Input and while Loops in Python

### Introduction paragraph (2-3 paragraphs):

Most programs are written to solve an end user’s problem. To do so, you usually need to get some information from the user.
In this article we will learn how to accept user input so your program can then work with it. When your program needs a name,you will be able to prompt the user for a name. When your program needs a list of names, you will be able to prompt the user for a series of names. To achieve this, we’ll use the input() function.We’ll also learn how to keep programs running as long as users want them to, so they can enter as much information as they need to; then, your program can work with that information. We will use Python’s while loop to keep programs running as long as certain conditions remain true.

With the ability to work with user input and the ability to control how long your programs run, you will be able to write fully interactive programs.

### Key takeaways: 

1.How the input() Function Works
2.Using int() to Accept Numerical Input
3.Introducing while Loops
4.Using a while Loop with Lists and Dictionaries

### References: 
N\A

### PR DESCRIPTION
**Pull request by Bonimum**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntx highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
  
This fixes #2340

### CODE REVIEW

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by Bonimum** [comment]
hey @paulodhiambo  any updates?

File: content/articles/user-input-and-while-loops-in-python/index.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
+As a programmer, many programs that you'll write will solve an end user's problem. You will always want to get some data from the user. This article will teach us how to accept user input and how to use while loops. While loops enable your programs to remain executing when certain conditions remain true.
+
+### Prerequisites
+1. Python installed on your computer.
+2. Basic Python knowledge.
+
+### How the input() function works
+The `input()` function halts your program and waits the user to key in some data. When Python receives the user's input, store it in the variable that you choose to work with.
```
Comment (paulodhiambo) [inline content/articles/user-input-and-while-loops-in-python/index.md:8]: Rewrite "The `input()` function halts your program and waits the user to key in some data." to "The `input()` function halts the execution of a program and waits for the user to key in some data."

File: content/articles/user-input-and-while-loops-in-python/index.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+```python
+name = input("Please enter your name: ")
+print(name)
+#output
+#Please enter your name: John
+#John
+```
+
+The `input()` function requires one argument, that is, the instruction you want the user to see. In this example, Python executes the first line and requests the user to enter his name. The program will halt and wait for the user to enter his name and continue after he presses **ENTER**. The name of the user is then loaded in the variable *name*, then `print(name)` displays the user's name back.
```
Comment (paulodhiambo) [inline content/articles/user-input-and-while-loops-in-python/index.md:20]: Rewrite "The `input()` function requires one argument" to "The `input()` function takes in one argument"

File: content/articles/user-input-and-while-loops-in-python/index.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+### Using int() to accept numerical input
+Any text the user enters using the `input()` function Python interprets it as a string. If your work is to print the input, the `input()` function works very well. But if you use the input as a number, an error arises. Consider the example below:
```
Comment (paulodhiambo) [inline content/articles/user-input-and-while-loops-in-python/index.md:23]: Rewrite "Any text the user enters using the `input()`..." to "Any text the user input using the `input()`..."

File: content/articles/user-input-and-while-loops-in-python/index.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+```
+
+Output:
+```
+Please enter your amount? 9000
+Your interest is : 
+504.0
+```
+
+When we enter 9000 in this example, Python interprets it as a string. The amount is then converted to an integer using the `int()` function. Now Python calculates the value of the interest.
```
Comment (paulodhiambo) [inline content/articles/user-input-and-while-loops-in-python/index.md:53]: Rewrite "When we enter 9000 in this example,..." to "When we input 9000 in this example,..."

File: content/articles/user-input-and-while-loops-in-python/index.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+#output
+#The count is: 10
+#The count is: 11
+#The count is: 12
+#The count is: 13
+#The count is: 14
+#The count is: 15
+```
+
+The *while* loop continues to execute if the value of *count* is equal to or less than 15.
```
Comment (paulodhiambo) [inline content/articles/user-input-and-while-loops-in-python/index.md:80]: Use back ticks for function names and keywords. Change "The *while* loop continues to execute if the value of *count* is equal to or less than 15." to "The `while` loop continues to execute if the value of `count` is equal to or less than 15."

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
@Bonimum amazing article on loops and input. Review the requested changes and ping when done.

**Comment by Bonimum** [comment]
> @Bonimum amazing article on loops and input. Review the requested changes and ping when done.

@paulodhiambo  changes requested done

Thank you

**Comment by paulodhiambo** [comment]
@Bonimum use [Hero](https://github.com/paulodhiambo/spring-amazon/blob/master/hero.png) as your hero image.

**Comment by Bonimum** [comment]
> @Bonimum use [Hero](https://github.com/paulodhiambo/spring-amazon/blob/master/hero.png) as your hero image.

@paulodhiambo Have added hero image

**Comment by paulodhiambo** [comment]
@Bonimum amazing article. @hectorkambow you can give it a final review.

**Review by louisefindlay23 — APPROVED** [review]
Great article, @Bonimum. Final review complete. 

Thanks @paulodhiambo for peer reviewing.

**Pull request merged**